p2p/server.go

```diff
@@ -367,6 +368,16 @@ func (srv *Server) RemovePeer(node *enode.Node) {
 		ch  chan *PeerEvent
 		sub event.Subscription
 	)
+
+	// Special case: sending a disconnect request with its own enode ID will reset the disconnect enode set
+	if node.ID() == srv.localnode.ID() {
+		srv.disconnectEnodeSetMu.Lock()
+		srv.disconnectEnodeSet = make(map[enode.ID]struct{})
+		srv.disconnectEnodeSetMu.Unlock()
+		srv.log.Debug("Reset disconnect enode set")
+		return
+	}
+
 	// Disconnect the peer on the main loop.
 	srv.doPeerOp(func(peers map[enode.ID]*Peer) {
 		srv.dialsched.removeStatic(node)
```

Comment:
we can move the clean up operation inside of this function, it will be executed in the server's main loop, so it will not have the concurrent access issue, so need the extra lock

Reply:
besides, why not use Magic EnodeID? I think it could be much easier for maintenance.

Reply:
My initial thought is that the ID is guaranteed to be unique because a node can't connect to itself as a peer. But this introduces an additional where a node operator has to call `admin_nodeInfo` or store the enode ID somewhere. Yeah, not too good for maintenance.

Already amended to use a magic enode ID instead.